client/src/components/Draftail/index.js

```diff
@@ -154,6 +154,21 @@ const initEditor = (selector, originalOptions, currentScript) => {
     field.draftailEditor = ref;
   };
 
+  const getChooserUrls = (type) => {
+    const chooserUrls = {};
+    const chooserUrl = originalOptions.chooserUrls;
+
+    if (type === 'LINK') {
+      chooserUrls.pageChooser = chooserUrl?.pageChooser;
+      chooserUrls.externalLinkChooser = chooserUrl?.externalLinkChooser;
+      chooserUrls.emailLinkChooser = chooserUrl?.emailLinkChooser;
+      chooserUrls.phoneLinkChooser = chooserUrl?.phoneLinkChooser;
+      chooserUrls.anchorLinkChooser = chooserUrl?.anchorLinkChooser;
+    }
+
+    return chooserUrls;
+  };
```

Comment:
I really think we should avoid this 'unique' handling for Links and instead pass the link entity data in just like all others.

Adding this override for LINK type hard-coded is risky as it could be confusing later as to why.

Instead, I have tested and think it should work that we use the same approach as all other features and prepare this in the entity inner options.

`wagtail/admin/wagtail_hooks.py`

```py
                "allowlist": {
                    # Keep pasted links with http/https protocol, and not-pasted links (href = undefined).
                    "href": "^(http:|https:|undefined$)",
                },
                "chooserUrls": {
                    "pageChooser": reverse_lazy("wagtailadmin_choose_page"),
                    "externalLinkChooser": reverse_lazy("wagtailadmin_choose_page_external_link"),
                    "emailLinkChooser": reverse_lazy("wagtailadmin_choose_page_email_link"),
                    "phoneLinkChooser": reverse_lazy("wagtailadmin_choose_page_phone_link"),
                    "anchorLinkChooser": reverse_lazy("wagtailadmin_choose_page_anchor_link"),
                }
            },
            js=[
                "wagtailadmin/js/page-chooser-modal.js",
            ],
```

Reply:
yes I agree and I removed "chooserUrls" hard-coded it is not the best approach.